tiledb/sm/filesystem/posix.cc

```diff
@@ -347,6 +347,26 @@ Status Posix::copy_file(
   return Status::Ok();
 }
 
+Status Posix::copy_dir(
+    const std::string& old_path, const std::string& new_path) {
+  create_dir(new_path);
```

Comment:
Check the return status here. `RETURN_NOT_OK(create_dir(new_path))` is sufficient. Do the same for the other routines here.